middlewares/middlewares.go

```diff
@@ -0,0 +1,43 @@
+package middlewares
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/TheDevExperiment/server/internal/utility/jwt"
+	"github.com/TheDevExperiment/server/router/models/auth"
+	"github.com/gin-gonic/gin"
+)
+
+const ContextKeyUserId = "UserId"
+const ContextKeyIsGuest = "IsGuest"
+
+/*
+if token is valid, attaches userId to the context.
+Note that this middleware simply verifies the signature only.
+It won't check if the Subject (in claim) is still an active user/not
+
+Those aspect can be handled by handler functions themselves for now.
+*/
+func JWTAuthMiddleware(c *gin.Context) {
+	var res auth.Response
+	bearerToken := strings.Split(
```

Comment:
is an index out of bounds possible here?

Reply:
yup!!